packages/headless/src/features/generated-answer/generated-answer-slice.ts

```diff
@@ -94,4 +96,9 @@ export const generatedAnswerReducer = createReducer(
           ...new Set(state.fieldsToIncludeInCitations.concat(action.payload)),
         ];
       })
+      .addCase(executeSearch.pending, (state) => handleResetAnswer(state))
+      .addCase(streamAnswer.pending, (state) => {
+        state.isLoading = true;
+        state.isStreaming = false;
+      })
```

Comment:
I think we need a new case here for when the stream answer is completed that sets `isLoading` to `false`.

For now we are reseting `isLoading` to false only  in the cases `updateMessage`, `updateCitations` and `updateError`.
If for example streamId is returned but no streaming is received, `isLoading`  will remain `true` causing an issue.

```suggestion
      .addCase(streamAnswer.pending, (state) => {
        state.isLoading = true;
        state.isStreaming = false;
      })
      .addCase(streamAnswer.fulfilled, (state) => {
        state.isLoading = false;
      })
```

Reply:
Unfortunately cannot do this, as `streamAnswer` is not truly "async": it exit/return in a synchronous manner when you dispatch it, and actually writes/stream the answer by dispatching other actions.

So, net net, is that the `loading` state would just become un-necessary from a UX standpoint (you'd never see the loading animation).

I think all scenarios are covered with `write/abort/close` on the stream, and for each of those the loading state is properly handled with a dispatch.